File: Atividades/Lista3/Java/LocadoraDeVeiculo.java
package Atividades.Lista3.Java;

import java.util.Scanner;

public class LocadoraDeVeiculo {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        String[] veiculoStr = input.nextLine().split(" ");
        String[] motoristaStr = input.nextLine().split(" ");
        String[] clienteStr = input.nextLine().split(" ");
        String[] chamadoStr = input.nextLine().split(" ");

        Veiculo veiculo = new Veiculo(veiculoStr[0], veiculoStr[1], veiculoStr[2], veiculoStr[3], Integer.parseInt(veiculoStr[4]));

        Motorista motorista = new Motorista(Integer.parseInt(motoristaStr[0]), motoristaStr[1], motoristaStr[2], motoristaStr[3]);

        Cliente cliente = new Cliente(Integer.parseInt(clienteStr[0]), clienteStr[3], clienteStr[4], clienteStr[1], clienteStr[2]);

        Chamado chamado = new Chamado(Integer.parseInt(chamadoStr[0]), chamadoStr[1], chamadoStr[2], chamadoStr[3], chamadoStr[4], Float.parseFloat(chamadoStr[5]), Float.parseFloat(chamadoStr[6]), Double.parseDouble(chamadoStr[7]), Double.parseDouble(chamadoStr[8]), Double.parseDouble(chamadoStr[9]), cliente, motorista, veiculo);

        System.out.printf("%d\n%s\n%s\n%s\n%s\n",
            chamado.getId(),
            chamado.getDestino(),
            chamado.getVeiculo().getPlaca(),
            chamado.getCliente().getRg(),
            chamado.getMotorista().getNome()
        );
        input.close();
    }
}
class Cliente {
    private int id;
    private String rg, cpf, nome, telefone;

    public Cliente(int id, String rg, String cpf, String nome, String telefone) {
        this.id = id;
        this.rg = rg;
        this.cpf = cpf;
        this.nome = nome;
        this.telefone = telefone;
    }

    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }
    public String getRg() {
        return rg;
    }
    public void setRg(String rg) {
        this.rg = rg;
    }
    public String getCpf() {
        return cpf;
    }
    public void setCpf(String cpf) {
        this.cpf = cpf;
    }
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }
    public String getTelefone() {
        return telefone;
    }
    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }
}

class Motorista {
    private int id;
    private String nome, telefone, cnh;

    public Motorista(int id, String nome, String telefone, String cnh) {
        this.id = id;
        this.nome = nome;
        this.telefone = telefone;
        this.cnh = cnh;
    }

    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }
    public String getNome() {
        return nome;
    }
    public void setNome(String nome) {
        this.nome = nome;
    }
    public String getTelefone() {
        return telefone;
    }
    public void setTelefone(String telefone) {
        this.telefone = telefone;
    }
    public String getCnh() {
        return cnh;
    }
    public void setCnh(String cnh) {
        this.cnh = cnh;
    }
}

class Veiculo {
    private String placa, marca, modelo, cor;
    private int ano;
    
    public Veiculo(String placa, String marca, String modelo, String cor, int ano) {
        this.placa = placa;
        this.marca = marca;
        this.modelo = modelo;
        this.cor = cor;
        this.ano = ano;
    }

    public String getPlaca() {
        return placa;
    }
    public void setPlaca(String placa) {
        this.placa = placa;
    }
    public String getMarca() {
        return marca;
    }
    public void setMarca(String marca) {
        this.marca = marca;
    }
    public String getModelo() {
        return modelo;
    }
    public void setModelo(String modelo) {
        this.modelo = modelo;
    }
    public String getCor() {
        return cor;
    }
    public void setCor(String cor) {
        this.cor = cor;
    }
    public int getAno() {
        return ano;
    }
    public void setAno(int ano) {
        this.ano = ano;
    }
}

class Chamado {
    private int id;
    private String data, tipo, origem, destino;
    private float horarioPartida, horarioRetorno;
    private double quilometragemIn, quilometragemFin, valorTotal;
    private Cliente cliente;
    private Motorista motorista;
    private Veiculo veiculo;

    public Chamado(int id, String data, String tipo, String origem, String destino, float horarioPartida, float horarioRetorno, double quilometragemIn, double quilometragemFin, double valorTotal, Cliente cliente, Motorista motorista, Veiculo veiculo) {
        this.id = id;
        this.data = data;
        this.tipo = tipo;
        this.origem = origem;
        this.destino = destino;
        this.horarioPartida = horarioPartida;
        this.horarioRetorno = horarioRetorno;
        this.quilometragemIn = quilometragemIn;
        this.quilometragemFin = quilometragemFin;
        this.valorTotal = valorTotal;
        this.cliente = cliente;
        this.motorista = motorista;
        this.veiculo = veiculo;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getOrigem() {
        return origem;
    }

    public void setOrigem(String origem) {
        this.origem = origem;
    }

    public String getDestino() {
        return destino;
    }

    public void setDestino(String destino) {
        this.destino = destino;
    }

    public float getHorarioPartida() {
        return horarioPartida;
    }

    public void setHorarioPartida(float horarioPartida) {
        this.horarioPartida = horarioPartida;
    }

    public float getHorarioRetorno() {
        return horarioRetorno;
    }

    public void setHorarioRetorno(float horarioRetorno) {
        this.horarioRetorno = horarioRetorno;
    }

    public double getQuilometragemIn() {
        return quilometragemIn;
    }

    public void setQuilometragemIn(double quilometragemIn) {
        this.quilometragemIn = quilometragemIn;
    }

    public double getQuilometragemFin() {
        return quilometragemFin;
    }

    public void setQuilometragemFin(double quilometragemFin) {
        this.quilometragemFin = quilometragemFin;
    }

    public double getValorTotal() {
        return valorTotal;
    }

    public void setValorTotal(double valorTotal) {
        this.valorTotal = valorTotal;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public Motorista getMotorista() {
        return motorista;
    }

    public void setMotorista(Motorista motorista) {
        this.motorista = motorista;
    }

    public Veiculo getVeiculo() {
        return veiculo;
    }

    public void setVeiculo(Veiculo veiculo) {
        this.veiculo = veiculo;
    }
}

File: Atividades/Lista_3/C++/locadoraDeVeiculo.cpp
#include <iostream>

using namespace std;

class Cliente {
private:
    int id;
    string rg, cpf, nome, telefone;

public:
    Cliente(int id, string rg, string cpf, string nome, string telefone)
        : id(id), rg(rg), cpf(cpf), nome(nome), telefone(telefone) {}

    int getId() {
        return id;
    }
    string getRg() {
        return rg;
    }
    string getCpf() {
        return cpf;
    }
    string getNome() {
        return nome;
    }
    string getTelefone() {
        return telefone;
    }
    void setId(int id) {
        this->id = id;
    }
    void setRg(string rg) {
        this->rg = rg;
    }
    void setCpf(string cpf) {
        this->cpf = cpf;
    }
    void setNome(string nome) {
        this->nome = nome;
    }
    void setTelefone(string telefone) {
        this->telefone = telefone;
    }

};

class Motorista {
private:
    int id;
    string nome, telefone, cnh;

public:
    Motorista(int id, string nome, string telefone, string cnh)
        : id(id), nome(nome), telefone(telefone), cnh(cnh) {}

    int getId() {
        return id;
    }
    string getNome() {
        return nome;
    }
    string getTelefone() {
        return telefone;
    }
    string getCnh() {
        return cnh;
    }
    void setId(int id) {
        this->id = id;
    }
    void setNome(string nome) {
        this->nome = nome;
    }
    void setTelefone(string telefone) {
        this->telefone = telefone;
    }
    void setCnh(string cnh) {
        this->cnh = cnh;
    }
};

class Veiculo {
private:
    string placa, marca, modelo, cor;
    int ano;
    
public:
    Veiculo(string placa, string marca, string modelo, string cor, int ano)
        : placa(placa), marca(marca), modelo(modelo), cor(cor), ano(ano)
    {}

    string getPlaca() {
        return placa;
    }
    string getMarca() {
        return marca;
    }
    string getModelo() {
        return modelo;
    }
    string getCor() {
        return cor;
    }
    int getAno() {
        return ano;
    }
    void setPlaca(string placa) {
        this->placa = placa;
    }
    void setMarca(string marca) {
        this->marca = marca;
    }
    void setModelo(string modelo) {
        this->modelo = modelo;
    }
    void setCor(string cor) {
        this->cor = cor;
    }
    void setAno(int ano) {
        this->ano = ano;
    }
};

class Chamado {
private:
    int id;
    string data, tipo, origem, destino;
    float horarioPartida, horarioRetorno;
    double quilometragemIn, quilometragemFin, valorTotal;
    Cliente cliente;
    Motorista motorista;
    Veiculo veiculo;

public:
    Chamado(int id, string data, string tipo, string origem, string destino, float horarioPartida, float horarioRetorno, double quilometragemIn, double quilometragemFin, double valorTotal, Cliente& cliente, Motorista& motorista, Veiculo& veiculo)
        : id(id), data(data), tipo(tipo), origem(origem), destino(destino), horarioPartida(horarioPartida), horarioRetorno(horarioRetorno), quilometragemIn(quilometragemIn), quilometragemFin(quilometragemFin), valorTotal(valorTotal), cliente(cliente), motorista(motorista), veiculo(veiculo) {}

    int getId() {
        return id;
    }

    void setId(int id) {
        this->id = id;
    }

    string getData() {
        return data;
    }

    void setData(string data) {
        this->data = data;
    }

    string getTipo() {
        return tipo;
    }

    void setTipo(string tipo) {
        this->tipo = tipo;
    }

    string getOrigem() {
        return origem;
    }

    void setOrigem(string origem) {
        this->origem = origem;
    }

    string getDestino() {
        return destino;
    }

    void setDestino(string destino) {
        this->destino = destino;
    }

    float getHorarioPartida() {
        return horarioPartida;
    }

    void setHorarioPartida(float horarioPartida) {
        this->horarioPartida = horarioPartida;
    }

    float getHorarioRetorno() {
        return horarioRetorno;
    }

    void setHorarioRetorno(float horarioRetorno) {
        this->horarioRetorno = horarioRetorno;
    }

    double getQuilometragemIn() {
        return quilometragemIn;
    }

    void setQuilometragemIn(double quilometragemIn) {
        this->quilometragemIn = quilometragemIn;
    }

    double getQuilometragemFin() {
        return quilometragemFin;
    }

    void setQuilometragemFin(double quilometragemFin) {
        this->quilometragemFin = quilometragemFin;
    }

    double getValorTotal() {
        return valorTotal;
    }

    void setValorTotal(double valorTotal) {
        this->valorTotal = valorTotal;
    }

    Cliente getCliente() {
        return cliente;
    }

    void setCliente(Cliente cliente) {
        this->cliente = cliente;
    }

    Motorista getMotorista() {
        return motorista;
    }

    void setMotorista(Motorista motorista) {
        this->motorista = motorista;
    }

    Veiculo getVeiculo() {
        return veiculo;
    }

    void setVeiculo(Veiculo veiculo) {
        this->veiculo = veiculo;
    }
};

int main() {
    int ano, id;
    string placa, marca, modelo, cor, nome, telefone, cnh, rg, cpf, data, tipo, origem, destino;
    float horarioPartida, horarioRetorno;
    double quilometragemIn, quilometragemFin, valorTotal;

    cin >> placa >> marca >> modelo >> cor >> ano;

    Veiculo veiculo(placa, marca, modelo, cor, ano);

    cin >> id >> nome >> telefone >> cnh;

    Motorista motorista(id, nome, telefone, cnh);

    cin >> id >> nome >> telefone >> rg >> cpf;

    Cliente cliente(id, rg, cpf, nome, telefone);

    cin >> id >> data >> tipo >> origem >> destino >> horarioPartida >> horarioRetorno >> quilometragemIn >> quilometragemFin >> valorTotal;

    Chamado chamado(id, data, tipo, origem, destino, horarioPartida, horarioRetorno, quilometragemIn, quilometragemFin, valorTotal, cliente, motorista, veiculo);

    cout << chamado.getId() << endl << chamado.getDestino() << endl << chamado.getVeiculo().getPlaca() << endl << chamado.getCliente().getRg() << endl << chamado.getMotorista().getNome() << endl;

    return 0;
}